ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh

```diff
@@ -205,31 +205,56 @@ function extract_migrated_caps() {
     done
 }
 
-# Get the capabilities which are introdcued for the 1st time in the target payload
-# By comparing two payloads at manifest level
-function extract_potential_new_caps() {
-    local source_dir="$1"
-    local target_dir="$2"
-    local target_cap_files target_file filename source_file target_cap
-    local candidate_new_caps=""
-
+# Get the new caps transfered from an existing operator on source verion
+# by removing those cap operators which is totally unknow on source version
+function get_transfered_new_caps() {
+    local source_dir="$1" target_dir="$2" candidate_new_caps="$3" found="no"
+    local cap target_cap_files target_file filename source_file target_cap crd_files cr_kind
+    
     # Check for missing environment variables
-    if [ -z "$source_dir" ] || [ -z "$target_dir" ]; then
+    if [ -z "$source_dir" ] || [ -z "$target_dir" ] || [[ -z "$candidate_new_caps" ]]; then
         echo "ERROR: Missing required arguments"
         return 1
     fi
-
-    target_cap_files=$(grep -rl "capability.openshift.io/name" ${target_dir})
-    for target_file in ${target_cap_files}; do
-        filename=$(basename "${target_file}")
-        source_file="$source_dir/$filename"
-
-        if [ ! -f "$source_file" ]; then
-            target_cap=$(grep 'capability.openshift.io/name:' "$target_file" | cut -d ':' -f2 | tr -d '"' | sort | uniq )
-            candidate_new_caps="${candidate_new_caps} ${target_cap}"
+    for cap in ${candidate_new_caps}; do
+        target_cap_files=$(grep -rl "capability.openshift.io/name: ${cap}$" ${target_dir} || true)
+        if [[ -z "$target_cap_files" ]]; then
+            echo "Did not find out $cap capacity manifest files in target payload"
```

Comment:
If the new_cap is not found in the target payload, it should not be expected, right? Should the test exit?

Reply:
Take `DeploymentConfig` as example, there is no any capacity manifest files can be found, so introduce this if block to continue.